### PR TITLE
Feature: Add form validation logic and CardSuccess component

### PR DESCRIPTION
**Pull request by GeorgeLR87**
- Created the `CardSuccess` component to display upon successful form submission.
- Added state `const [isSubmitted, setIsSubmitted] = useState(false)` to toggle between form and success component upon validation completion.
- Implemented the state `const [errors, setErrors] = useState<FormErrors>({})` in the Card component to handle form input validation.
- Updated color palette across the project for a consistent UI design.
- Developed a validation helper function `validateForm` to check for empty fields and validate input formats for cardholder name, card number, expiration date, and CVC.
- Ensured form validation provides clear error messages for blank fields and incorrect card formats, including checks for non-numeric characters and 16-digit card number validation.

image
image
image
image

### CODE REVIEW

File: src/components/CardSuccess.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const CardSuccess = () => {
+  return (
```
Comment (DaroMacs) [inline src/components/CardSuccess.tsx:16]: Remove unnecesary return make it implicit

File: src/components/Dates/Dates.tsx [hunk 2]
```diff
@@ -10,21 +10,33 @@ type ExpDatesInputProps = {
...
 dispatch: Dispatch<FormActions>;
+errors?: {
+    expMonth?: string;
+    expYear?: string;
+};
 }
 
-const Dates = ({ state, dispatch }: ExpDatesInputProps) => {
+const Dates = ({ state, dispatch, errors }: ExpDatesInputProps) => {
     return (
```
Comment (DaroMacs) [inline src/components/Dates/Dates.tsx]: unnecesary return

File: src/helpers/helpers.ts [hunk 3]
```diff
@@ -23,4 +26,31 @@ export const formatYear = (year: string): string => {
...
+    } else if (!/^\d{16}$/.test(state[EFormData.CardNumber].replace(/\s/g, ''))) {
+        errors[EFormData.CardNumber] = "Must be 16 digits";
+    }
+    if (!state[EFormData.ExpMonth]) {
+        errors[EFormData.ExpMonth] = "Can't be blank";
+    }
+    if (!state[EFormData.ExpYear]) {
+        errors[EFormData.ExpYear] = "Can't be blank";
+    }
+    if (!state[EFormData.CVC]) {
```
Comment (DaroMacs) [inline src/helpers/helpers.ts:51]: this CVC validation should check if the input is 3 digits cos CVC are made of 3 digits always.

https://www.loom.com/share/21123672a1b24272a48c52a44e8ff431

**Comment by DaroMacs** [comment]
I would change the error message to something like Required instead of cant be blank cos in the date the text makes 2 rows and looks bad

image

File: src/components/CardSuccess.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            strokeLinejoin="miter"
+            d="M5 13l4 4L19 7"
+          />
+        </svg>
+      </div>
+
+      <h2 className="text-xl  text-deep-violet mt-4 tracking-widest">THANK YOU!</h2>
+      <p className="text-sm text-purplish-grey mt-2">We’ve added your card details</p>
+
+      <button
```
Comment (DaroMacs) [inline src/components/CardSuccess.tsx:38]: Add a onClick method make sure it cleans the form and sets the state of isSubmitted to false so you show the form again

**Comment by GeorgeLR87** [comment]
I have made the requested changes and aligned them with the project requirements outlined in Figma.

image
image

**Review by DaroMacs — APPROVED** [review]

**Pull request merged**